skimage/filters/ridges.py

```diff
@@ -106,8 +108,278 @@ def _check_sigmas(sigmas):
     return sigmas
 
 
+def gaussian_filter_with_FFT_kernel(input, sigma, order=0, output=None,
```

Comment:
I think we would be better off placing this and the 1D function it uses in `skimage/filters/gaussian.py`.  Ideally, this would go upstream in SciPy itself, but I think implementing it here first is reasonable. Even if we got it into SciPy 1.8 it would still be at least 2 years before scikit-image would have 1.8 as a minimum requirement, so we would need a local copy for awhile anyway!

The current `skimage.filters.gaussian` does not expose an order parameter, but I still think it makes sense to group these FFT-based Gaussians in that same file.

Reply:
Right, it definitely makes sense to group it based on its category, and I can imagine it may have a wider use. I'm open to nicer name suggestion than `gaussian_filter_with_truncated_FFT_kernel` :-)